testing/python/transform/test_tilelang_transform_im2col_fallback.py

```
@@ -65,6 +65,50 @@ def test_im2col_uses_tma_on_hopper():
+
+    @T.prim_func
+    def kern(
+        data: T.Tensor((N, H, W, C), T.float16),
+        out: T.Tensor((block_M, block_K), T.float16),
+    ):
+        with T.Kernel(1, threads=128):
+            sh = T.alloc_shared((block_M, block_K), T.float16)
+            T.im2col(data, sh, 0, 0, K, 1, 1, 1)
+            T.copy(sh, out)
+
+    with pytest.raises(Exception, match="im2col cannot lower.*16-byte multiple"):
```

Ruff (0.16.2)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)